### ISSUE TITLE
Generate getter unconditionally inserts $0 (snippet position marker)

### ISSUE THREAD

**Issue by azriel91** [rust-lang/rust-analyzer#14161]
[URLO thread](https://users.rust-lang.org/t/rust-analyzer-generates-function-names-with-0/89275)

Heya, when I use "generate getter" on the `value` field using Rust analyzer:

```rust
#[derive(Clone, Debug)]
pub struct MyStruct {
    value: u32,
}
```

The following code gets generated (notice the `$0` before the function name):

```rust
impl MyStruct {
    pub fn $0value(&self) -> u32 {
        self.value
    }
}
```

Possibly [this line](https://github.com/rust-lang/rust-analyzer/blob/89b994049147e6ed947c8ee5a29547be448874a0/crates/ide-assists/src/handlers/generate_getter.rs#L183) should only insert the `$0` if `ctx.config.snippet_cap.is_some()` -- based on [this other part](https://github.com/rust-lang/rust-analyzer/blob/89b994049147e6ed947c8ee5a29547be448874a0/crates/ide-assists/src/handlers/generate_getter.rs#L223-L226) further down.

I haven't found a way to see what capabilities sublime text is reporting to LSP, in case we need to see that.

---

**rust-analyzer version**: `rust-analyzer 0.0.0 (89b994049 2023-02-14)`
**rustc version**: `rustc 1.69.0-nightly (0416b1a6f 2023-02-14)`
**relevant settings**:

* sublime text 4147
* [LSP](https://github.com/sublimelsp/LSP) v1.22.0
* [LSP-rust-analyzer](https://github.com/sublimelsp/LSP-rust-analyzer) v1.3.1

**Comment by Veykril** [comment]
Ye that is missing the `snippet_cap` check
